pkg/logql/engine.go

```diff
@@ -265,7 +263,7 @@ func (q *query) Exec(ctx context.Context) (logqlmodel.Result, error) {
 	queueTime, _ := ctx.Value(httpreq.QueryQueueTimeHTTPHeader).(time.Duration)
 
 	statResult := statsCtx.Result(time.Since(start), queueTime, q.resultLength(data))
-	statResult.Log(level.Debug(spLogger))
+	statResult.LogWithSpan(sp)
```

Comment:
I wonder about a little bit different approach where we leave the original stats.Log() method and use that to print the stats in metrics.go.

But here do: `sp.LogKV(statResult.KVList()...)`

Reply:
i like yours style more but fyi we use `LogWithSpan` at other places. I'll try using `LogKV` there too then.

Reply:
pushed https://github.com/grafana/loki/pull/13255/commits/0c9570ada806d7ab5b99f3fa3ebc5a0c159e1bbb addressing it